src/middlewared/middlewared/plugins/smb_sharesec.py

```diff
@@ -0,0 +1,409 @@
+from middlewared.schema import Bool, Dict, List, Str, Int
+from middlewared.service import (accepts, filterable, private, periodic, CRUDService)
+from middlewared.service_exception import CallError
+from middlewared.utils import run, filter_list
+from middlewared.plugins.smb import SMBCmd
+
+import enum
+import errno
+import os
+import re
+
+RE_SHAREACLENTRY = re.compile(r"^ACL:(?P<ae_who_sid>.+):(?P<ae_type>.+)\/0x0\/(?P<ae_perm>.+)$")
+
+
+class SIDType(enum.IntEnum):
+    """
+    Defined in MS-SAMR (2.2.2.3) and lsa.idl
+    Samba's group mapping database will primarily contain SID_NAME_ALIAS entries (local groups)
+    """
+    NONE = 0
+    USER = 1
+    DOM_GROUP = 2
+    DOMAIN = 3
+    ALIAS = 4
+    WELL_KNOWN_GROUP = 5
+    DELETED = 6
+    INVALID = 7
+    UNKNOWN = 8
+    COMPUTER = 9
+    LABEL = 10
+
+
+class ShareSec(CRUDService):
+
+    class Config:
+        namespace = 'smb.sharesec'
+
+    @private
+    async def _parse_share_sd(self, sd, options=None):
```

Comment:
You don't need to mark them with @private if they start with underscore, they won't be visible outside this class